File: Contest on Code Forces/Harbour Space Scholarship/main1.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if(haystack==needle):
            return 0
        elif(needle in haystack):
            index = 0
            flag = False
            for i in range(len(haystack)):
                if(haystack[i]==needle[index]):
                    flag = True
                    index+=1
                else:
                    flag=False
                    index = 0

                if(index == len(needle)-1 and flag == True):
                    return i+2-len(needle)
                    
        else:
            return -1

File: Contest on Code Forces/Harbour Space Scholarship/main1.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if(haystack==needle)
        {
            return 0;
        }
        else if (haystack.size()==0)
        {
            return 0;
        }
        else{
            int index = 0;
            bool flag = false;
            for (int i=0; i<haystack.size(); i++)
            {
                if(haystack[i]==needle[index])
                {
                    index+=1;
                    flag = true;
                }
                else{
                    flag = false;
                    index = 0;
                }
                if(index == needle.size()-1 && flag==true)
                {
                    return i+2-needle.size();
                }
            }
            return -1;
        }
    }
};